src/Core/ServiceProvider.php

```diff
@@ -33,6 +33,14 @@
  */
 abstract class ServiceProvider extends AbstractServiceProvider implements BootableServiceProviderInterface
 {
+    /**
+     * List of ids of services this provider provides.
+     *
+     * @var array<string>
+     * @see ServiceProvider::provides()
+     */
+    protected $services = [];
```

Comment:
Isn't the old property name [provides](https://container.thephpleague.com/3.x/service-providers/)? Could we use that so the upgrade is even easier?

Reply:
Reverted to the old property name. I missed that fact that the properly already existed, just the new method was added.

Reply:
Nice. This will help make upgrading much simpler. :clap: